akka-http2-support/src/main/scala/akka/http/impl/engine/http2/FrameEvent.scala

```diff
@@ -4,37 +4,62 @@
 
 package akka.http.impl.engine.http2
 
-import akka.http.impl.engine.http2.Protocol.FrameType
-import akka.http.impl.engine.http2.Protocol.SettingIdentifier
+import akka.http.impl.engine.http2.Http2Protocol.FrameType
+import akka.http.impl.engine.http2.Http2Protocol.SettingIdentifier
 import akka.util.ByteString
 
-sealed trait FrameEvent
+sealed trait FrameEvent {
+  def flags: ByteFlag
+  def hasFlag(f: ByteFlag): Boolean = f.isSet(flags)
+}
 sealed trait StreamFrameEvent extends FrameEvent {
   def streamId: Int
 }
 
 final case class DataFrame(
-  streamId:  Int,
-  endStream: Boolean,
-  payload:   ByteString) extends StreamFrameEvent
+  flags:    ByteFlag,
+  streamId: Int,
+  payload:  ByteString) extends StreamFrameEvent {
+  def endStream = Http2Protocol.Flags.END_STREAM.isSet(flags)
+}
 
 final case class HeadersFrame(
+  flags:               ByteFlag,
```

Comment:
Another con would be that flags don't map 1-to-1 to the events. We have `SettingsFrame` and `SettingsAckFrame` which will both generate a SETTINGS frame one with a payload and the other with a flag set. And for HEADERS and DATA we may have padding which we currently don't model in the events because it doesn't carry any semantics.